examples/decode_sensor_binary_log.py

```diff
@@ -107,9 +109,8 @@ def unpack_array(cls, file: IO[Any]):
         if array_size <= cls.MAX_ARRAY_LENGTH:
             return file.read(array_size)
 
-    @classmethod
-    def unpack_string(cls, file: IO[Any]):
-        return cls.unpack_array(file).decode('UTF-8')
+    def unpack_string(self, file: IO[Any]):
```

Comment:
Shouldn't this self charge occur in the first commit ?

Reply:
Nope - the first commit is fixing a bug where the `recover` method added failed byte tracking, which was stored in the instance. That meant that `self` was required for `recover` and for `unpack_message` (which uses `recover`), but there's no effect on the other class methods.

This commit adds instance-based encoding tracking depending on the initial header read, which then affects `unpack_string` for subsequent parsing, which is why it then needs to use the encoding stored in `self`, so can no longer be just a class method.